### PR TITLE
Respect runtime types in ASAs

### PR DESCRIPTION
**Pull request by la4ezar**
**Description**
When processing ASAs we need to respect the runtime types in the formation template and assign to the formation only the runtimes (or runtime contexts of that runtimes) that are of the type provided in the formation template.
If we have a runtime(even if the runtime has owner=true) with runtime contexts, only runtime contexts must be assigned to the formation.

Changes proposed in this pull request:
- Respect runtime types in ASAs
- Unit tests
- E2E test

**Related issue(s)**

- https://github.com/kyma-incubator/compass/pull/2434

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: tests/director/tests/formation_api_test.go [hunk 1]
```diff
@@ -563,6 +571,193 @@ func TestTenantFormationFlow(t *testing.T) {
...
+		kymaRuntime := registerKymaRuntime(t, ctx, subscriptionConsumerSubaccountID, kymaRtmInput)
+		defer fixtures.CleanupRuntime(t, ctx, certSecuredGraphQLClient, subscriptionConsumerSubaccountID, &kymaRuntime)
+
+		// Register kyma formation template
+		kymaFormationTmplName := "kyma-formation-template-name"
+		createFormationTemplate(t, ctx, subscriptionConsumerSubaccountID, "kyma", kymaFormationTmplName, conf.KymaRuntimeTypeLabelValue, graphql.ArtifactTypeEnvironmentInstance)
+
+		// Register provider formation template
+		providerFormationTmplName := "provider-formation-template-name"
+		createFormationTemplate(t, ctx, subscriptionProviderSubaccountID, "provider", providerFormationTmplName, conf.KymaRuntimeTypeLabelValue, graphql.ArtifactTypeSubscription)
```
Comment (DimitarPetrov) [inline tests/director/tests/formation_api_test.go:667]: maybe runtime type here should be conf.SubscriptionProviderAppNameValue

File: tests/director/tests/formation_api_test.go [hunk 2]
```diff
@@ -563,6 +571,193 @@ func TestTenantFormationFlow(t *testing.T) {
...
+	t.Logf("Unassign tenant %s from formation %s...", objectID, formationName)
+	unassignReq := fixtures.FixUnassignFormationRequest(objectID, "TENANT", formationName)
+	var formation graphql.Formation
+	err := testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, unassignReq, &formation)
+	require.NoError(t, err)
+	require.Equal(t, formationName, formation.Name)
+	t.Logf("Successfully unassigned tenant %s from formation %s", objectID, formationName)
+}
+
+func createFormationTemplate(t *testing.T, ctx context.Context, tenantID, prefix, formationTemplateName, runtimeType string, runtimeArtifactKind graphql.ArtifactType) {
```
Comment (DimitarPetrov) [inline tests/director/tests/formation_api_test.go:741]: templates are global, no tenant id is needed

File: components/director/pkg/graphql/graphqlizer/graphqlizer.go [hunk 3]
```diff
@@ -531,6 +531,14 @@ func (g *Graphqlizer) TemplateValueInputToGQL(in graphql.TemplateValueInput) (st
 	}`)
 }
 
+// FormationInputToGQL missing godoc
```
Comment (DimitarPetrov) [inline components/director/pkg/graphql/graphqlizer/graphqlizer.go:534]: It is better to add go doc for new exported methods

File: tests/director/tests/formation_api_test.go [hunk 4]
```diff
@@ -563,6 +571,193 @@ func TestTenantFormationFlow(t *testing.T) {
...
+
+	formationTmplGQLInput, err := testctx.Tc.Graphqlizer.FormationTemplateInputToGQL(formationTmplInput)
+	require.NoError(t, err)
+	formationTmplRequest := fixtures.FixCreateFormationTemplateRequest(formationTmplGQLInput)
+
+	ft := graphql.FormationTemplate{}
+	t.Logf("Creating formation template with name: %q", formationTemplateName)
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, formationTmplRequest, &ft)
+	require.NoError(t, err)
+	defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, ft.ID)
```
Comment (DimitarPetrov) [inline tests/director/tests/formation_api_test.go:758]: This needs to be outside otherwise the template will be deleted immediately

Comment (PetarTodorovv) [reply tests/director/tests/formation_api_test.go]: yep, I found it when I was validating the test locally

File: tests/director/tests/formation_api_test.go [hunk 5]
```diff
@@ -563,10 +571,187 @@ func TestTenantFormationFlow(t *testing.T) {
...
+			return subscription.GetSubscriptionJobStatus(t, httpClient, subJobStatusURL, subscriptionToken) == subscription.JobSucceededStatus
+		}, subscription.EventuallyTimeout, subscription.EventuallyTick)
+		t.Logf("Successfully created subscription between consumer with subaccount id: %q and tenant id: %q, and provider with name: %q, id: %q and subaccount id: %q", subscriptionConsumerSubaccountID, subscriptionConsumerTenantID, providerRuntime.Name, providerRuntime.ID, subscriptionProviderSubaccountID)
+
+		// Register kyma runtime
+		kymaRtmInput := fixtures.FixRuntimeRegisterInput("kyma-runtime")
+		kymaRuntime := registerKymaRuntime(t, ctx, subscriptionConsumerSubaccountID, kymaRtmInput)
+		defer fixtures.CleanupRuntime(t, ctx, certSecuredGraphQLClient, subscriptionConsumerSubaccountID, &kymaRuntime)
+
+		// Register kyma formation template
```
Comment (la4ezar) [inline tests/director/tests/formation_api_test.go:660]: Change this comment to `// Create kyma formation template`. Also, change following comments too.

**Review by PetarTodorovv — APPROVED** [review]

**Review by desislavaa — APPROVED** [review]

**Pull request merged**